lib/firmware/nano.c

```diff
@@ -0,0 +1,74 @@
+#include <string.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include "keepkey/firmware/nano.h"
+#include "trezor/crypto/blake2b.h"
+
+static uint8_t const NANO_BLOCK_HASH_PREAMBLE[32] = {
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x06,
+};
+
+bool nano_path_mismatched(const CoinType *coin,
+                          const uint32_t *address_n,
+                          const uint32_t address_n_count)
+{
+    // m/44' : BIP44-like path
+    // m / purpose' / bip44_account_path' / account'
+    bool mismatch = false;
+    mismatch |= address_n[0] != (0x80000000 + 44);
```

Comment:
```suggestion
    mismatch |= address_n_count > 0 && address_n[0] != (0x80000000 + 44);
```

Reply:
Yes :man_facepalming: .. amended the commit